scripts/preinstall.js

```diff
@@ -0,0 +1,51 @@
+/********************************************************************************
+ * Copyright (C) 2020 TypeFox and others.
+ *
+ * This program and the accompanying materials are made available under the
```

Comment:
I don't think we need this script. My concern was about end products, not about building from this repo. In this repo we should specify proper engines in root package.json.

Reply:
IIUC this script is a some kind of migration warning, to get people on board with the upgrade.

If we directly bump the engine version for node, clients will get an error as soon as they pull the new Theia packages while still using node 10.

So we either still allow both node 10 and 12 with this warning to move to 12, or we switch to 12 only and break people? Maybe I'm missing something.

Reply:
- this PR does not disallow Node.js 10
- this check is not used by downstream clients
- this check is used for this repo, but there is already a check from yarn based on engines

Reply:
> [Unless the user has set the engine-strict config flag, this field is advisory only and will only produce warnings when your package is installed as a dependency.](https://docs.npmjs.com/files/package.json#engines)

Ok I thought it would have crashed package installation by default, my bad.

Reply:
> s a some kind of migration warning, to get people on board with the upgrade.

Yes, that was my intention. Thanks!

----
We decided to:
 - not to be so restrictive with the Node.js version. Node.js 10.x should work the same as 12.x. So we will update the `node` constraint to `">=10.11.0 <13"`,
 - we remove the pre-install script, and won't log any warnings when building/installing Theia in an inappropriate env,
 - users can still ignore the `engines` constraint with the [`--ignore-engines`](https://classic.yarnpkg.com/en/docs/cli/install/#toc-yarn-install-ignore-engines) flag.

I will take care of these 👆